include/sys/ring_buffer.h

```diff
@@ -128,7 +128,7 @@ struct ring_buf {
 #define RING_BUF_DECLARE(name, size8) \
 	BUILD_ASSERT(size8 < RING_BUFFER_MAX_SIZE,\
 		RING_BUFFER_SIZE_ASSERT_MSG); \
-	static uint8_t __noinit _ring_buffer_data_##name[size8]; \
+	static uint8_t __noinit _ring_buffer_data_##name[size8] __aligned(sizeof(int)); \
```

Comment:
Hmm, as we need alignment here - are you sure that `__aligned(sizeof(int))` is enough on all platforms, especially 64 bit ones? Probably it would be better to use something like `uintptr_t` instead of `int` here?

PS: do we use `__aligned(sizeof(x))`  instead of `__alignof__` intentionally here?

Reply:
actually, it looks like an issue with noinit section in arc platform. If i add
```
uint8_t __noinit t;
void main(void)
{
  printk("%d\n", t);
}
```
it hangs on `qemu_arc_hs`

Reply:
Adding `__aligned(sizeof(int))` fixes it.  Given that i will remove this alignment from ring buffer and will open an issue.